File: packages/core-go/muxed/decode.go
package muxed

import (
	"encoding/binary"
	"fmt"

	"github.com/stellar-address-kit/core-go/address"
)

func DecodeMuxed(mAddress string) (string, string, error) {
	muxedAccount, err := strkey.DecodeMuxedAccount(mAddress)
	if err != nil {
		return "", "", err
	}

	baseG, err := muxedAccount.AccountID()
	if err != nil {
		return "", 0, err
	}

	return baseG, strconv.FormatUint(muxedAccount.ID(), 10), nil
}

File: packages/core-ts/src/muxed/decode.ts
import { MuxedAccount } from "@stellar/stellar-sdk";
import { MuxedResult } from "./types";

/**
 * Decodes a Stellar muxed address (M-address).
 *
 * @param {string} mAddress - The muxed address to decode.
 * @returns {baseG: string, id: bigint}
 */
export function decodeMuxed(mAddress: string): MuxedResult {
  const muxed = MuxedAccount.fromAddress(mAddress, "0");
  return {
    baseG: muxed.baseAccount().accountId(),
    id: BigInt(muxed.id()),
  };
}
